client/src/index.ts

```diff
@@ -304,12 +304,17 @@ export class RestRepository<T> extends Repository<T> {
   }
 
   async request(method: string, path: string, body?: unknown) {
-    const { databaseURL, apiKey } = this.client.options;
+    const { databaseURL, apiKey, branch: branchStrategy } = this.client.options;
     if (!databaseURL || !apiKey) {
       throw new Error('Options databaseURL and apiKey are required');
     }
 
-    const resp: Response = await this.fetch(`${databaseURL}${path}`, {
+    const branch = await getBranchFromStrategy(branchStrategy);
+    if (!branch) {
+      throw new Error('Option branch is required');
+    }
```

Comment:
This could be a cost consuming operation, should we memoize it?

Reply:
I think we can just remember this value in a new field in `XataClient`. Then we'd do something like this:

```ts
this.branch = this.branch || await getBranchFromStrategy(branchStrategy);
```